src/views/visual/visual-view-graph.js

```diff
@@ -1,43 +1,105 @@
-import { css, customElement, html, LitElement, property } from 'lit-element';
+import { customElement, html, LitElement, property } from 'lit-element';
 
 import { getDataForVisual } from '../../api/actions';
 import '../utility/LoadingSpinner';
 import { getGoogleGraphOptions } from './visualViewUtils';
 
+import styles from './visual-view-graph.styles';
+
 @customElement('visual-view-graph')
 export class VisualViewGraph extends LitElement {
-  @property({ type: Array }) graphData;
-  @property({ type: String }) language;
   @property({ type: String }) searchItem;
   @property({ type: String }) selectedCollections;
+  @property({ type: Number }) pageSize;
+  @property({ type: Function }) setSelection;
+
+  @property({ type: Array }) graphData;
+  @property({ type: String }) language;
+  @property({ type: Number }) currentPage = 0;
+  @property({ type: Number }) totalPages;
   @property({ type: String }) chartHeight;
   @property({ type: String }) fetchLoading;
   @property({ type: String }) fetchError;
 
   static get styles() {
-    return [
-      css`
-        #parallels-chart {
-          font: 13px sans-serif;
-          padding: 30px;
-          width: 93vw;
-        }
-      `,
-    ];
+    return [styles];
   }
 
   updated(_changedProperties) {
-    console.log('visual view updated. ', _changedProperties);
     _changedProperties.forEach((oldValue, propName) => {
       if (
         ['searchItem', 'selectedCollections'].includes(propName) &&
         !this.fetchLoading
       ) {
         this.fetchData();
       }
+
+      if (['currentPage'].includes(propName) && !this.fetchLoading) {
+        this.adjustChartHeight();
+      }
     });
   }
 
+  paginateGraphData(graphData) {
+    let paginatedGraphData = [];
+    let currentPage = [];
+    let alreadyFoundTexts = [];
+    let count = 0;
+    let entryCount = 0;
+    graphData.forEach(entry => {
+      currentPage.push(entry);
+      entryCount += 1;
+      if (!alreadyFoundTexts.includes(entry[0])) {
+        alreadyFoundTexts.push(entry[0]);
+        count += 1;
+      }
+      if (count >= this.pageSize || entryCount == graphData.length) {
+        paginatedGraphData.push(currentPage);
+        currentPage = [];
+        count = 0;
+      }
+    });
+    return paginatedGraphData;
+  }
+
+  decreaseCurrentPage() {
+    if (this.currentPage > 0) {
+      this.currentPage -= 1;
+    }
+  }
+
+  increaseCurrentPage() {
+    if (this.currentPage < this.totalPages - 1) {
+      this.currentPage += 1;
+    }
+  }
+
+  createPageDisplay() {
+    if (this.totalPages <= 1) {
+      return;
+    }
+    let pages = [];
+    for (let i = 0; i < this.totalPages; i++) {
+      let currentClass = 'element';
+      if (i == this.currentPage) {
+        currentClass += ' active';
+      }
+      // prettier-ignore
+      pages.push(
+        html`<span class="${currentClass}" @click="${function() {this.currentPage = i;}}">${i + 1}</span>`
```

Comment:
maybe an arrow function would be nicer? `@click="${() => this.currentPage = i}"`